src/c++/perf_analyzer/infer_data_manager.cc

```diff
@@ -85,111 +185,19 @@ InferDataManager::InitInferDataOutput(
 
 cb::Error
 InferDataManager::UpdateInputs(
-    const int stream_index, const int step_index, InferData& infer_data)
+    const size_t thread_id, const int stream_index, const int step_index,
+    InferData& infer_data)
 {
   // Reset inputs for this inference request
   infer_data.valid_inputs_.clear();
 
   for (const auto& input : infer_data.inputs_) {
-    RETURN_IF_ERROR(input->Reset());
```

Comment:
Why no more reset here? And why keep it for shm?

Reply:
The purpose of the function is to populate valid_inputs_ with the correct data.

old code would delete all data in the inputs_, copy the new data from the dataloader into inputs_, and then point valid_inputs_ to those inputs_ that are valid.

New data uses the _inputs as a reference to the names and types of inputs, but directly points _valid_inputs to the pre-generated data (which is aware of its validity) skipping the need to reset or modify _inputs.